### DISCUSSION TITLE
Icons based on data / dynamic icon

### DISCUSSION THREAD

**Discussion by black-forest-digital** [livewire/flux#416] [Feature Requests] [answered]
Having the name of the heroicon in my database, I'd like to choose the icon dynamically. 

`<flux:icon.{{ $phase->icon }} />`

Currently I see it in the html generated but it's not rendering.

Is there a way to achieve it or is it a feature request?

Best, Philipp

**Comment by jeffchown** [comment]
You can use the same method used in the `vendor/livewire/flux/stubs/resources/views/flux/icon/index.blade.php` file:

`<x-dynamic-component :component="'flux::icon.' . $phase->icon" />`

**Reply by joshhanley** [reply, marked as answer]
Or you can just use the icon component 😁

```blade
<flux:icon :name="$phase->icon" />
```

**Reply by jeffchown** [reply]
Nice @joshhanley.  Even better! 👍

**Reply by black-forest-digital** [reply]
Thanks! 
Might be interesting for others in the docs?

**Reply by joshhanley** [reply]
Whoops was meant to do that. I've just submitted a PR to add this.
